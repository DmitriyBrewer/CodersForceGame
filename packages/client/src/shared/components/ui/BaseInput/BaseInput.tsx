import { InputBase } from '@mui/material'

const BaseInput = InputBase

export default BaseInput
