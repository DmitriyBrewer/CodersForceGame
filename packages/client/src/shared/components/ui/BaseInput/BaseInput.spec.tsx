import { render, screen } from '@testing-library/react'

import '@testing-library/jest-dom'
import BaseInput from './BaseInput'

describe('<BaseInput />', () => {
  const handleClick = jest.fn()

  it('Snapshot is correct', () => {
    const { asFragment } = render(<BaseInput onClick={handleClick} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('Renders correctly with default props', () => {
    render(<BaseInput />)
    const textbox = screen.getByRole('textbox')

    expect(textbox).toBeInTheDocument()
    expect(textbox).toMatchSnapshot()
  })
})
