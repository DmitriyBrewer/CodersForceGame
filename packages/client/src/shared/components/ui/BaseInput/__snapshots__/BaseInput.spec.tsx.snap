// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseInput /> Renders correctly with default props 1`] = `
<input
  class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
  type="text"
  value=""
/>
`;

exports[`<BaseInput /> Snapshot is correct 1`] = `
<DocumentFragment>
  <div
    class="MuiInputBase-root MuiInputBase-colorPrimary css-1le28he-MuiInputBase-root"
  >
    <input
      class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
